### DISCUSSION TITLE
Support for creating Azure Entra roles

### DISCUSSION THREAD

**Discussion by ThomasShih** [cyrilgdn/terraform-provider-postgresql#438] [Q&A] [answered]
Hi all!

Just wondering if there is any recommended way of creating an postgresql role with azure entra capabilities?

Ideally, there is an entra enabled flag on the `posgresql_role` resource that I can specify is an entra config, but that doesn't exist. Alternatively, if there is a way to execute raw sql, I can grant a existing role that has been created entra login abilities via [this](https://registry.terraform.io/providers/cyrilgdn/postgresql/latest/docs/resources/postgresql_role).

Has anyone else had to resolve this? If so, I would love some ideas/suggestions on a clean implimitation.

**Comment by Teabeats** [comment, marked as answer]
this PR: https://github.com/cyrilgdn/terraform-provider-postgresql/pull/365 will make this possible by adding labels as described here: https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/how-to-manage-azure-ad-users#enable-microsoft-entra-authentication-for-an-existing-postgresql-role-using-sql - hopefully @cyrilgdn is able to have a look at it soon
